### PR TITLE
[e2e] Support dual-stack - ensureloadbalancer & nsg

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind testing

#### What this PR does / why we need it:
[e2e] Support dual-stack - ensureloadbalancer & nsg
#### Which issue(s) this PR fixes:

Fixes #
related #814
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/network/network_security_group.go [hunk 1]
```diff
@@ -380,38 +413,72 @@ func validateSharedSecurityRuleExists(nsgs []aznetwork.SecurityGroup, ips []stri
...
 			continue
 		}
-		for _, securityRule := range *nsg.SecurityRules {
-			utils.Logf("Checking security rule %q", pointer.StringDeref(securityRule.Name, ""))
-			if securityRule.Access == aznetwork.SecurityRuleAccessDeny &&
-				strings.EqualFold(pointer.StringDeref(securityRule.DestinationAddressPrefix, ""), ip) &&
-				strings.HasSuffix(pointer.StringDeref(securityRule.Name, ""), "deny_all") &&
-				strings.EqualFold(pointer.StringDeref(securityRule.SourceAddressPrefix, ""), "*") {
-				return true
+		utils.Logf("Chechking nsg %q", pointer.StringDeref(nsg.Name, ""))
```
Comment (jwtty) [inline tests/e2e/network/network_security_group.go:463]: nit: Checking

File: tests/e2e/network/network_security_group.go [hunk 2]
```diff
@@ -380,38 +413,72 @@ func validateSharedSecurityRuleExists(nsgs []aznetwork.SecurityGroup, ips []stri
...
 			continue
 		}
-		for _, securityRule := range *nsg.SecurityRules {
-			utils.Logf("Checking security rule %q", pointer.StringDeref(securityRule.Name, ""))
-			if securityRule.Access == aznetwork.SecurityRuleAccessAllow &&
-				strings.EqualFold(pointer.StringDeref(securityRule.DestinationAddressPrefix, ""), ip) &&
-				strings.HasSuffix(pointer.StringDeref(securityRule.Name, ""), ipRangesSuffix) &&
-				strings.EqualFold(pointer.StringDeref(securityRule.SourceAddressPrefix, ""), sourceAddressPrefix) {
-				return true
+		utils.Logf("Chechking nsg %q", pointer.StringDeref(nsg.Name, ""))
```
Comment (jwtty) [inline tests/e2e/network/network_security_group.go:424]: nit: Checking

File: tests/e2e/network/network_security_group.go [hunk 3]
```diff
@@ -360,6 +392,7 @@ func validateSharedSecurityRuleExists(nsgs []aznetwork.SecurityGroup, ips []stri
 		if nsg.SecurityRules == nil {
 			continue
 		}
+		utils.Logf("Chechking nsg %q", pointer.StringDeref(nsg.Name, ""))
```
Comment (jwtty) [inline tests/e2e/network/network_security_group.go:395]: nit: Checking

File: tests/e2e/network/network_security_group.go [hunk 4]
```diff
@@ -330,27 +347,42 @@ var _ = Describe("Network security group", Label(utils.TestSuiteLabelNSG), func(
...
 	for _, nsg := range nsgs {
 		if nsg.SecurityRules == nil {
 			continue
 		}
-		for _, securityRule := range *nsg.SecurityRules {
-			utils.Logf("Checking security rule %q", pointer.StringDeref(securityRule.Name, ""))
-			if strings.EqualFold(pointer.StringDeref(securityRule.DestinationAddressPrefix, ""), ip) && strings.EqualFold(pointer.StringDeref(securityRule.DestinationPortRange, ""), port) {
-				utils.Logf("Found target security rule")
-				return true
+		utils.Logf("Chechking nsg %q", pointer.StringDeref(nsg.Name, ""))
```
Comment (jwtty) [inline tests/e2e/network/network_security_group.go:369]: nit: Checking

File: tests/e2e/network/network_security_group.go (hunk 1 above)
Comment (lzhecheng) [reply tests/e2e/network/network_security_group.go]: Updated.

File: tests/e2e/network/network_security_group.go (hunk 2 above)
Comment (lzhecheng) [reply tests/e2e/network/network_security_group.go]: Updated.

File: tests/e2e/network/network_security_group.go (hunk 3 above)
Comment (lzhecheng) [reply tests/e2e/network/network_security_group.go]: Updated.

File: tests/e2e/network/network_security_group.go (hunk 4 above)
Comment (lzhecheng) [reply tests/e2e/network/network_security_group.go]: Updated.

**Pull request merged**